### PR TITLE
Volume pricing - price per item

### PR DESCRIPTION
**Pull request by eugenekasimov**
### PR Summary: 

This PR adds a price per unit next to quantity input field on PDP for products that have volume pricing. 

### Why are these changes introduced?

Fixes #0.

### Things I still need to work on here
- ~~Change hardcoded text to locales~~
- ~~Isolate some scope of JS that causes an errors in console now.~~ 
- ~~For some reason, when you switch to another variant and add to cart 2+ items, the qty input field is not reseting to the min qty.~~
- ~~Probably we need to add loading spinner~~

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |

### Visual impact on existing themes

If a product has volume pricing you will see a price per item next to quantity input field on PDP. 

### Testing steps/scenarios

- [ ] Login as B2B
- [ ] Go to a product that has volume pricing and make sure there is a price per item next to  quantity input field
- [ ] Start changing qty in quantity input field back and forward and make sure the ppi changes as well
- [ ] Switch to another variant and make sure it still works
- [ ] Start adding an item to cart buy pressing add to cart button, make sure it updates ppi after you added something to cart
- [ ] Try different cart type. Switch to drawer cart. Add an item to cart and start changing qty in cart. Make sure ppi changes accordingly. 

### Demo links

- [Editor](https://admin.shopify.com/store/os2-demo/themes/140204081174/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by edmund-teh** [comment]
Sorry, I just realized you're still working on this @eugenekasimov!
I'm sure you're already aware of this bug: https://screenshot.click/23-49-qg18n-pk7v9.mp4

Please feel free to ignore me and continue working on this... 😆

**Comment by eugenekasimov** [comment]
>

**Comment by eugenekasimov** [comment]
> Sorry, I just realized you're still working on this @eugenekasimov! I'm sure you're already aware of this bug: https://screenshot.click/23-49-qg18n-pk7v9.mp4
> 
> Please feel free to ignore me and continue working on this... 😆

Nice catch, I fixed it.

**Review by scottgmeadows — APPROVED** [review]
UX is looking nice :)

File: sections/main-product.liquid [hunk 1]
```diff
@@ -235,36 +235,60 @@
...
+                    assign curentQtyForVolumePricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
+
+                  -%}
+                  {%- if product.quantity_price_breaks_configured? -%}
+                    <div class='price-per-item'>
+                      {%- if curentQtyForVolumePricing  < volumePricingArray.last.minimum_quantity -%}
+                        <span>at</span>
+                        <span>{{ product.selected_or_first_available_variant.price | money_with_currency }}/ea</span>
+                      {%- else -%}
+                        {%- for price_break in volumePricingArray -%}
```
Comment (sofiamatulis) [inline sections/main-product.liquid:282]:
```suggestion
                        {%- for price_break in volume_pricing_array -%}
```

File: sections/main-product.liquid [hunk 2]
```diff
@@ -235,36 +235,60 @@
...
+                    </button>
+                  </quantity-input>
+                  {%- liquid
+                    assign volumePricingArray = product.selected_or_first_available_variant.quantity_price_breaks | sort: 'price'
+                    assign curentQtyForVolumePricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
+
+                  -%}
+                  {%- if product.quantity_price_breaks_configured? -%}
+                    <div class='price-per-item'>
+                      {%- if curentQtyForVolumePricing  < volumePricingArray.last.minimum_quantity -%}
```
Comment (sofiamatulis) [inline sections/main-product.liquid:278]:
```suggestion
                      {%- if current_qty_volume_pricing  < volume_pricing_array.last.minimum_quantity -%}
```

File: sections/main-product.liquid [hunk 3]
```diff
@@ -235,36 +235,60 @@
...
+                    />
+                    <button class="quantity__button no-js-hidden" name="plus" type="button">
+                      <span class="visually-hidden">
+                        {{- 'products.product.quantity.increase' | t: product: product.title | escape -}}
+                      </span>
+                      {% render 'icon-plus' %}
+                    </button>
+                  </quantity-input>
+                  {%- liquid
+                    assign volumePricingArray = product.selected_or_first_available_variant.quantity_price_breaks | sort: 'price'
```
Comment (sofiamatulis) [inline sections/main-product.liquid:272]:
```suggestion
                    assign volume_pricing_array = product.selected_or_first_available_variant.quantity_price_breaks | sort: 'price'
```

File: sections/main-product.liquid [hunk 4]
```diff
@@ -235,36 +235,60 @@
...
+                    assign curentQtyForVolumePricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
```
Comment (sofiamatulis) [inline sections/main-product.liquid:273]:
```suggestion
                    assign current_qty_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
```

File: assets/global.js [hunk 5]
```diff
@@ -157,28 +157,88 @@ class QuantityInput extends HTMLElement {
     super();
     this.input = this.querySelector('input');
     this.changeEvent = new Event('change', { bubbles: true });
-
+    if (this.input.hasAttribute('data-volume-pricing')) {
+      this.isVolumePricing = this.querySelector('input[name="quantity"]').dataset.volumePricing;
```
Comment (sofiamatulis) [inline assets/global.js:161]:
```suggestion
      this.isVolumePricing = this.querySelector('input[name="quantity"]').dataset.volumePricing === 'true';
```

File: assets/global.js [hunk 6]
```diff
@@ -157,28 +157,88 @@ class QuantityInput extends HTMLElement {
...
     );
   }
 
   quantityUpdateUnsubscriber = undefined;
+  updatePricePerItemUnsubscriber = undefined;
 
   connectedCallback() {
     this.validateQtyRules();
     this.quantityUpdateUnsubscriber = subscribe(PUB_SUB_EVENTS.quantityUpdate, this.validateQtyRules.bind(this));
+    if (this.isVolumePricing === 'true') {
```
Comment (sofiamatulis) [inline assets/global.js:175]:
```suggestion
    if (this.isVolumePricing) {
```

File: assets/global.js [hunk 7]
```diff
@@ -157,28 +157,88 @@ class QuantityInput extends HTMLElement {
...
       this.quantityUpdateUnsubscriber();
     }
+    if (this.updatePricePerItemUnsubscriber) {
+      this.updatePricePerItemUnsubscriber();
+    }
   }
 
   onInputChange(event) {
     this.validateQtyRules();
+    if (this.isVolumePricing === 'true') this.updatePricePerItem();
```
Comment (sofiamatulis) [inline assets/global.js:194]:
```suggestion
    if (this.isVolumePricing) this.updatePricePerItem();
```

**Review by sofiamatulis — COMMENTED** [review]
Added a few comments 👌 Looking good

Ill re-test when we have the spinner/translations and look a bit more at the code

File: assets/constants.js [hunk 8]
```diff
@@ -5,4 +5,5 @@ const PUB_SUB_EVENTS = {
   quantityUpdate: 'quantity-update',
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/constants.js:5]: Okay, I expected this question :). I thought that it makes sense to add the spinner only for cases when we request updated data through API because that is the case when we potentially may have a delay. If you try to add to cart or using cart-drawer change qty in cart you will see the spinner.

Does it make sense?

File: sections/main-product.liquid [hunk 9]
```diff
@@ -235,36 +235,79 @@
...
+                            >
+                              <circle class="path" fill="none" stroke-width="6" cx="33" cy="33" r="30"></circle>
+                            </svg>
+                          </span>
+                        </span>
+                        <div class='price-per-item'>
+                          <span>{{ 'products.product.volume_pricing.at_price' | t }} </span>
+                          {%- if current_qty_for_volume_pricing  < volume_pricing_array.last.minimum_quantity -%}
+                            {%- assign varian_price = product.selected_or_first_available_variant.price | money_with_currency -%}
+                            <span>{{ 'products.product.volume_pricing.price_each' | t: price: varian_price }}</span>
```
Comment (sofiamatulis) [inline sections/main-product.liquid:296]:
```suggestion
                            <span>{{variant_price }}/ each</span>
```

File: sections/main-product.liquid [hunk 10]
```diff
@@ -235,36 +235,79 @@
...
+                              xmlns="http://www.w3.org/2000/svg"
+                            >
+                              <circle class="path" fill="none" stroke-width="6" cx="33" cy="33" r="30"></circle>
+                            </svg>
+                          </span>
+                        </span>
+                        <div class='price-per-item'>
+                          <span>{{ 'products.product.volume_pricing.at_price' | t }} </span>
+                          {%- if current_qty_for_volume_pricing  < volume_pricing_array.last.minimum_quantity -%}
+                            {%- assign varian_price = product.selected_or_first_available_variant.price | money_with_currency -%}
```
Comment (sofiamatulis) [inline sections/main-product.liquid:295]:
```suggestion
                            {%- assign variant_price = product.selected_or_first_available_variant.price | money_with_currency -%}
```

**Comment by eugenekasimov** [comment]
Okay team, let me summarize my thoughts. I also noticed that there is a bit of a misunderstanding of terminology, so I attached a quick explanation video. 

1. I'm not going to add the spinner when we change price by interacting with the input field because there is no network request.
2. We make the spinner every time you press the add to cart button or change the quantity in the cart because these make network requests. 
3. There is a known behaviour that we see the spinner each time we interact with the cart even when the price doesn't change. Although it may look not ideal and I personally agree with that I don't think that should be a blocker for this PR. Also, due to complexity, it's going to be quite time consuming to try to change this behaviour. I think we should address it in the future. 
https://screenshot.click/28-46-faf8q-a160g.mp4

**Comment by sofiamatulis** [comment]
> I think I'm okay with this, given our tight timelines and that there is no network request when changing the price by interacting with the input field.
> 
> But, is this something that could be a fast follow? Or should it be a blocker? @sofiamatulis

@edmund-teh If UX is ok with this, I agree it can be a follow up. So the follow up behaviour would be:

- Only show spinner when price changes (by adding to cart **or** playing with the picker

MVP version
- Showing the spinner when buyer adds to cart (price changing or not(
- Not showing the spinner at all when the price changes.

cc @nicklepine 

Would that match variant list? 🤔 Because variant list each time you click on the picker you add to cart or remove to cart (which is a network request). That means that the spinner shows every time

File: assets/product-info.js [hunk 11]
```diff
@@ -59,7 +59,7 @@ if (!customElements.get('product-info')) {
 
       fetchQuantityRules() {
         if (!this.currentVariant || !this.currentVariant.value) return;
-        this.querySelector('.quantity__rules-cart .loading-overlay').classList.remove('hidden');
+        this.querySelectorAll('.quantity__rules-cart .loading-overlay').forEach((overlay) => overlay.classList.remove('hidden'));
```
Comment (KaichenWang) [inline assets/product-info.js:62]: Showing loading spinner could also be handled by the new `price-per-item` component instead

**Review by KaichenWang — COMMENTED** [review]
I think the overall logic is good! I would suggest moving your logic to a new custom HTML component. The `quantity-input` component is meant to be more generic and used throughout the theme and may not always need to display a price per item; so it would be good to separate the logic.

File: sections/main-product.liquid [hunk 12]
```diff
@@ -235,36 +235,79 @@
...
+                      {% render 'icon-plus' %}
+                    </button>
+                  </quantity-input>
+                  {%- liquid
+                    assign volume_pricing_array = product.selected_or_first_available_variant.quantity_price_breaks | sort: 'price'
+                    assign current_qty_for_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
+
+                  -%}
+                  {%- if product.quantity_price_breaks_configured? -%}
+                    <label class="price-per-item__label form__label" for="Price-per-item-{{ section.id }}">
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply sections/main-product.liquid:277]: We don't need it here 👍

File: locales/en.default.json [hunk 13]
```diff
@@ -142,7 +142,9 @@
       "volume_pricing": {
         "title": "Volume Pricing",
         "note": "Volume pricing available",
-        "minimum": "{{ quantity }}+"
+        "minimum": "{{ quantity }}+",
+        "at_price": "at",
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply locales/en.default.json:146]: Honestly, I couldn't figure a better way. We don't want to add "at" to "price_each": "{{ price }}/ea" because as far as I understand we use "price_each" for volume-pricing and we don't want to see "at" there 🤔
Screenshot 2023-06-29 at 10 17 13 AM

File: assets/global.js [hunk 14]
```diff
@@ -157,28 +157,107 @@ class QuantityInput extends HTMLElement {
...
+    const volumePricing = document.querySelector('volume-pricing[id^="Volume-"]');
+    this.qtyPricePairs = [];
+
+    volumePricing.querySelectorAll('li').forEach(li => {
+      const qty = parseInt(li.querySelector('span:first-child').textContent);
+      const price = (li.querySelector('span:not(:first-child):last-child').textContent);
+      this.qtyPricePairs.push([qty, price]);
+    });
+
+    this.qtyPricePairs.reverse();
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/global.js:260]: That's a good idea. Thanks.

File: assets/global.js [hunk 15]
```diff
@@ -206,6 +205,103 @@ class QuantityInput extends HTMLElement {
 
 customElements.define('quantity-input', QuantityInput);
 
+class PricePerItem extends HTMLElement {
```
Comment (sofiamatulis) [inline assets/global.js:208]: Can you move this to its own file? We wont need it for every page. This way we dont load it every time

File: assets/product-form.js [hunk 16]
```diff
@@ -40,6 +40,8 @@ if (!customElements.get('product-form')) {
         }
         config.body = formData;
 
+        publish(PUB_SUB_EVENTS.cartUpdateStarted);
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/product-form.js:43]: That how it was previously. Kai suggested to move it to price-per-item component just to keep all about price per item in one place. I agree with that approach, it makes more sense to me. I don't see a big difference and have a strong preference on it. If everyone is okay I can move this add/remove to the sources of the events.

File: locales/en.default.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (KaichenWang) [reply locales/en.default.json:147]: They can be 2 different strings
```
"price_at_each": "at {{ price }}/ea",
"price_each": "{{ price }}/ea"
```

[the remark this answers is not in the document]
Comment (eugenekasimov) [reply locales/en.default.json:147]: That's not actually working for me. The way I update the price per item is I use an array in JS where I add data from liquid which already looks like `$100.00 CAD/ea`.

File: assets/price-per-item.js [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+        let pricePerItem = pair[1];
+        const pricePerItemCurrent = document.querySelector('.price-per-item span:last-child');
+        pricePerItemCurrent.innerHTML = pricePerItem;
+        break;
+      }
+    }
+  }
+
+  updateInputOnProductPage() {
+    this.input.value = Number(this.input.min);
```
Comment (KaichenWang) [inline assets/price-per-item.js:70]: Why is this needed?

Is `this.input` the quantity input? Wouldn't this mess with the `min` set by the `setQuantityBoundries()` function?

File: sections/main-product.liquid [hunk 18]
```diff
@@ -235,36 +236,75 @@
...
+                      <span class="visually-hidden">
+                        {{- 'products.product.quantity.decrease' | t: product: product.title | escape -}}
+                      </span>
+                      {% render 'icon-minus' %}
+                    </button>
+                    <input
+                      class="quantity__input"
+                      type="number"
+                      name="quantity"
+                      data-variant-id="{{ product.selected_or_first_available_variant.id }}"
```
Comment (KaichenWang) [inline sections/main-product.liquid:251]: `data-variant-id` and  `data-volume-pricing` could be moved to the `<price-per-item>` element?

File: assets/global.js [hunk 19]
```diff
@@ -1034,6 +1033,10 @@ class VariantSelects extends HTMLElement {
         const volumePricingSource = html.getElementById(
           `Volume-${this.dataset.originalSection ? this.dataset.originalSection : this.dataset.section}`
         );
+
+        const pricePerItemDestination = document.querySelector(`price-per-item .price-per-item`);
```
Comment (KaichenWang) [inline assets/global.js:1037]: We need to query by a unique id. Otherwise we'll run into issues with quick buy, because it's possible that multiple `price-per-item` elements are on the page.

See the examples in the same file that leverage `this.dataset.originalSection` and `this.dataset.section`

File: assets/price-per-item.js [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+  updatePricePerItem(updatedCartQuantity) {
+    this.getVolumePricingArray();
+
+    const enteredQty = parseInt(this.input.value);
+    this.currentQtyForVolumePricing = this.getCartQuantity(updatedCartQuantity) + enteredQty;
+
+    for (let pair of this.qtyPricePairs) {
+      if (this.currentQtyForVolumePricing >= pair[0]) {
+        let pricePerItem = pair[1];
+        const pricePerItemCurrent = document.querySelector('.price-per-item span:last-child');
```
Comment (KaichenWang) [inline assets/price-per-item.js:62]:
```suggestion
        const pricePerItemCurrent = this.querySelector('.price-per-item span:last-child');
```
Also, could we give the element a class instead of querying for generic `span:last-child`?

File: assets/price-per-item.js [hunk 21]
```diff
@@ -0,0 +1,96 @@
+class PricePerItem extends HTMLElement {
+  constructor() {
+    super();
+    this.input = this.querySelector('input');
```
Comment (KaichenWang) [inline assets/price-per-item.js:4]: Which input are we trying to query for here? I would try to query for something less generic. I think you could add a unique id to the input in Liquid and use that to query using `document.querySelector`

File: assets/price-per-item.js [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+    this.variantId = this.querySelector('input[name="quantity"]').dataset.variantId;
```
Comment (KaichenWang) [inline assets/price-per-item.js:5]: Is this `this.querySelector('input[name="quantity"]')` the same element as `this.input`?

File: assets/price-per-item.js [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+    this.input.addEventListener('change', this.onInputChange.bind(this));
```
Comment (KaichenWang) [inline assets/price-per-item.js:6]: We don't have to do this in this PR, but it would be better to add your event listener to an pub/sub. The quantity input publishes an event when its input value changes and the `price-per-item` can subscribe to that event.

File: assets/price-per-item.js [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+
+  updateInputOnProductPage() {
+    this.input.value = Number(this.input.min);
+  }
+
+  getCartQuantity(updatedCartQuantity) {
+    if (updatedCartQuantity || updatedCartQuantity === 0) {
+      return updatedCartQuantity;
+    } else {
+      const cartQuantity = parseInt(document.querySelector('input[name="quantity"]').dataset.cartQuantity);
```
Comment (KaichenWang) [inline assets/price-per-item.js:77]: Which input are we querying for here? There could be multiple inputs that match this on the page.

File: assets/price-per-item.js [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+      return cartQuantity;
+    }
+  }
+
+  getVolumePricingArray() {
+    const volumePricing = document.querySelector('volume-pricing[id^="Volume-"]');
```
Comment (KaichenWang) [inline assets/price-per-item.js:83]: 1 - There could be multiple volume-pricing tables on the page when quick buy is enabled.

2 - We're re-running this every time the quantity input value changes, which is expensive. We should cache this data somehow or obtain if some other way.

File: sections/main-product.liquid [hunk 26]
```diff
@@ -235,36 +236,75 @@
...
-                      max="{{ product.selected_or_first_available_variant.quantity_rule.max }}"
-                    {% endif %}
-                    step="{{ product.selected_or_first_available_variant.quantity_rule.increment }}"
-                    value="{{ product.selected_or_first_available_variant.quantity_rule.min }}"
-                    form="{{ product_form_id }}"
-                  />
-                  <button class="quantity__button no-js-hidden" name="plus" type="button">
-                    <span class="visually-hidden">
-                      {{- 'products.product.quantity.increase' | t: product: product.title | escape -}}
+                <price-per-item>
```
Comment (KaichenWang) [inline sections/main-product.liquid:239]: I don't think you need to nest the `quantity-input` inside the `price-per-item`. You could keep them separate and query the quantity input from the DOM using a unique id.

File: assets/product-form.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/product-form.js:43]: cc @KaichenWang thoughts?

[the remark this answers is not in the document]
Comment (KaichenWang) [reply assets/product-form.js:43]: I think this is fine. Without the `cartUpdateStarted` event, the `price-per-item` component is unaware of when an item is being added to the cart and won't know when to start showing its loader.

File: assets/price-per-item.js (hunk 21 above)
Comment (eugenekasimov) [reply assets/price-per-item.js:4]: The input already has `id="Quantity-{{ section.id }}"` I added `data-section-id="{{ section.id }}"` to the component and now I can query it `this.input = document.getElementById(`Quantity-${this.dataset.sectionId}`);`

File: assets/product-form.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/product-form.js:43]: Can't we just add the loader to the `product-form` when the request is sent? Im just trying to be mindful of not creating new events if we dont need to

File: assets/price-per-item.js (hunk 22 above)
Comment (eugenekasimov) [reply assets/price-per-item.js]: Actually I found a bug that was happening when you switched variant because I never actually updated variant ID in my logic. Fixed.

File: assets/price-per-item.js (hunk 25 above)
Comment (eugenekasimov) [reply assets/price-per-item.js:85]: 1 - I cannot imagine how this is possible.

File: assets/product-form.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/product-form.js:43]: We can but in this case we will also need to add it to cart.js. In my opinion a new event is not that harmful and we can use less code.

File: locales/en.default.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply locales/en.default.json:147]: @eugenekasimov Wouldnt creating two strings like what Kai suggested work?

[the remark this answers is not in the document]
Comment (eugenekasimov) [reply locales/en.default.json:147]: not really, I explained why in my previous post 👆

File: assets/price-per-item.js [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+        break;
+      }
+    }
+  }
+
+  getCartQuantity(updatedCartQuantity) {
+    if (updatedCartQuantity || updatedCartQuantity === 0) {
+      return updatedCartQuantity;
+    } else {
+      const cartQuantity = parseInt(this.input.dataset.cartQuantity);
```
Comment (sofiamatulis) [inline assets/price-per-item.js:81]:
```suggestion
      return parseInt(this.input.dataset.cartQuantity);
```

File: assets/price-per-item.js [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+      return cartQuantity;
```
Comment (sofiamatulis) [inline assets/price-per-item.js:82]:
```suggestion
```

File: locales/en.default.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply locales/en.default.json:147]: Okay, Kai suggested me another approach to keep my current logic and it seems working.

File: locales/en.default.json [hunk 29]
```diff
@@ -144,6 +144,7 @@
         "title": "Volume Pricing",
         "note": "Volume pricing available",
         "minimum": "{{ quantity }}+",
+        "price_at_each": "at {{ price }}/ea",
```
Comment (sofiamatulis) [inline locales/en.default.json:147]: Can we request translations now? 👀

File: assets/price-per-item.js [hunk 30]
```diff
@@ -0,0 +1,109 @@
...
+
+      updatePricePerItem(updatedCartQuantity) {
+        // updatedCartQuantity is undefined when qty is updated on product page. We need to sum entered qty and current qty in cart.
+        if (updatedCartQuantity === undefined) {
+          const enteredQty = parseInt(this.input.value);
+          this.currentQtyForVolumePricing = this.getCartQuantity(updatedCartQuantity) + enteredQty;
+        // updatedCartQuantity is not undefined when qty is updated in cart. We need to sum qty in cart and min qty for product.
+        } else {
+          this.currentQtyForVolumePricing = this.getCartQuantity(updatedCartQuantity) + parseInt(this.input.min);
+        }
```
Comment (sofiamatulis) [inline assets/price-per-item.js:75]:
```suggestion
 const enteredQty = parseInt(this.input.value);

this.currentQtyForVolumePricing = updatedCartQuantity ? this.getCartQuantity(updatedCartQuantity) + parseInt(this.input.min) : this.getCartQuantity(updatedCartQuantity) + enteredQty;
```

File: assets/price-per-item.js [hunk 31]
```diff
@@ -0,0 +1,109 @@
...
+          }
+        }
+      }
+
+      getCartQuantity(updatedCartQuantity) {
+        if (updatedCartQuantity || updatedCartQuantity === 0) {
+          return updatedCartQuantity;
+        } else {
+          return parseInt(this.input.dataset.cartQuantity);
+        }
```
Comment (sofiamatulis) [inline assets/price-per-item.js:92]:
```suggestion
const newCartQty = if (updatedCartQuantity || updatedCartQuantity === 0) ? updatedCartQuantity : parseInt(this.input.dataset.cartQuantity);

return newCartQty
```

File: assets/price-per-item.js (hunk 30 above)
Comment (sofiamatulis) [reply assets/price-per-item.js:75]: I think you could try

`this.currentQtyForVolumePricing = updatedCartQuantity === undefined ? (this.getCartQuantity(updatedCartQuantity) + parseInt(this.input.min)) : (this.getCartQuantity(updatedCartQuantity) + enteredQty);`

Comment (eugenekasimov) [reply assets/price-per-item.js:75]: this should work :)

File: assets/price-per-item.js [hunk 32]
```diff
@@ -0,0 +1,103 @@
...
+
+        // updatedCartQuantity is undefined when qty is updated on product page. We need to sum entered qty and current qty in cart.
+        // updatedCartQuantity is not undefined when qty is updated in cart. We need to sum qty in cart and min qty for product.
+        this.currentQtyForVolumePricing = updatedCartQuantity === undefined ? this.getCartQuantity(updatedCartQuantity) + enteredQty : this.getCartQuantity(updatedCartQuantity) + parseInt(this.input.min);
+
+        for (let pair of this.qtyPricePairs) {
+          if (this.currentQtyForVolumePricing >= pair[0]) {
+            let pricePerItem = pair[1];
+            const pricePerItemCurrent = document.querySelector(`price-per-item[id^="Price-Per-Item-${this.dataset.sectionId}"] .price-per-item`);
+            pricePerItemCurrent.innerHTML = pricePerItem;
```
Comment (sofiamatulis) [inline assets/price-per-item.js:78]:
```suggestion
            pricePerItemCurrent.innerHTML = pair[1];
```

File: assets/price-per-item.js [hunk 33]
```diff
@@ -0,0 +1,103 @@
...
+      updatePricePerItem(updatedCartQuantity) {
+        const enteredQty = parseInt(this.input.value);
+
+        // updatedCartQuantity is undefined when qty is updated on product page. We need to sum entered qty and current qty in cart.
+        // updatedCartQuantity is not undefined when qty is updated in cart. We need to sum qty in cart and min qty for product.
+        this.currentQtyForVolumePricing = updatedCartQuantity === undefined ? this.getCartQuantity(updatedCartQuantity) + enteredQty : this.getCartQuantity(updatedCartQuantity) + parseInt(this.input.min);
+
+        for (let pair of this.qtyPricePairs) {
+          if (this.currentQtyForVolumePricing >= pair[0]) {
+            let pricePerItem = pair[1];
```
Comment (sofiamatulis) [inline assets/price-per-item.js:76]:
```suggestion
```

File: assets/price-per-item.js [hunk 34]
```diff
@@ -0,0 +1,103 @@
...
+            const pricePerItemCurrent = document.querySelector(`price-per-item[id^="Price-Per-Item-${this.dataset.sectionId}"] .price-per-item`);
+            pricePerItemCurrent.innerHTML = pricePerItem;
+            break;
+          }
+        }
+      }
+
+      getCartQuantity(updatedCartQuantity) {
+        const newCartQty = (updatedCartQuantity || updatedCartQuantity === 0) ? updatedCartQuantity : parseInt(this.input.dataset.cartQuantity);
+        return newCartQty
```
Comment (sofiamatulis) [inline assets/price-per-item.js:86]:
```suggestion
return (updatedCartQuantity || updatedCartQuantity === 0) ? updatedCartQuantity : parseInt(this.input.dataset.cartQuantity);
```

File: assets/price-per-item.js [hunk 35]
```diff
@@ -0,0 +1,103 @@
...
+          this.getVolumePricingArray();
+        });
+        this.cartUpdateStartedUnsubscriber = subscribe(PUB_SUB_EVENTS.cartUpdateStarted, (event) => {
+          if (event.variantId !== this.variantId) return;
+          this.querySelector('.loading-overlay').classList.remove('hidden');
+        });
+
+        this.updatePricePerItemUnsubscriber = subscribe(PUB_SUB_EVENTS.cartUpdate, (response) => {
+          this.querySelector('.loading-overlay').classList.add('hidden');
+          // Item was added to cart via product page
```
Comment (sofiamatulis) [inline assets/price-per-item.js:31]: Do we need these comments? 👀

File: assets/price-per-item.js [hunk 36]
```diff
@@ -0,0 +1,103 @@
...
+      getCartQuantity(updatedCartQuantity) {
+        const newCartQty = (updatedCartQuantity || updatedCartQuantity === 0) ? updatedCartQuantity : parseInt(this.input.dataset.cartQuantity);
+        return newCartQty
+      }
+
+      getVolumePricingArray() {
+        const volumePricing = document.getElementById(`Volume-${this.dataset.sectionId}`);
+        this.qtyPricePairs = [];
+
+        volumePricing.querySelectorAll('li').forEach(li => {
```
Comment (sofiamatulis) [inline assets/price-per-item.js:93]: I wonder if we should target for the class instead of `li`? 🤔

**Review by sofiamatulis — APPROVED** [review]
Works well! Added small nitpicks

File: assets/cart.js [hunk 37]
```diff
@@ -6,6 +6,8 @@ class CartRemoveButton extends HTMLElement {
       event.preventDefault();
       const cartItems = this.closest('cart-items') || this.closest('cart-drawer-items');
       cartItems.updateQuantity(this.dataset.index, 0);
+
```
Comment (sofiamatulis) [inline assets/cart.js:9]: This will be the wrong price when JS is off (variant change and qty amount):

https://screenshot.click/06-42-ojxav-eyfja.png

I think we hide it for now since we show it in the dropdown the price (the same way we hide the table for now and well rethink it in no JS)

cc @KaichenWang @edmund-teh

File: assets/price-per-item.js [hunk 38]
```diff
@@ -0,0 +1,103 @@
+if (!customElements.get('price-per-item')) {
+  customElements.define(
+    'price-per-item',
+    class PricePerItem extends HTMLElement {
+      constructor() {
+        super();
+        this.input = document.getElementById(`Quantity-${this.dataset.sectionId}`);
+        this.variantId = document.getElementById(`Price-Per-Item-${this.dataset.sectionId}`).dataset.variantId;
```
Comment (KaichenWang) [inline assets/price-per-item.js:8]: I think this is equivalent: 

```suggestion
        this.variantId = this.dataset.variantId;
```

File: assets/price-per-item.js [hunk 39]
```diff
@@ -0,0 +1,103 @@
...
+
+        this.getVolumePricingArray();
+      }
+
+      updatePricePerItemUnsubscriber = undefined;
+      cartUpdateStartedUnsubscriber = undefined;
+      variantIdChangedUnsubscriber = undefined;
+
+      connectedCallback() {
+        // Update varinatId if variant is switched on product page
```
Comment (KaichenWang) [inline assets/price-per-item.js:19]:
```suggestion
        // Update variantId if variant is switched on product page
```

**Review by KaichenWang — COMMENTED** [review]
I'm not sure it's caused by this PR, but the Quick Add is causing a JS error. I'm still trying to figure out why this is happening.

Screenshot 2023-07-06 at 5 49 15 PM 1

File: assets/price-per-item.js (hunk 35 above)
Comment (eugenekasimov) [reply assets/price-per-item.js:31]: I find them helpful at least now when we are going to work with this code for featured product for example.

**Comment by eugenekasimov** [comment]
> I'm not sure it's caused by this PR, but the Quick Add is causing a JS error. I'm still trying to figure out why this is happening.
> 
> Screenshot 2023-07-06 at 5 49 15 PM 1

I found it and fixed. I missed `"` in one place.
